### DISCUSSION TITLE
I am new to wix toolset. You want to create an Insataller that meets the following requirements.

### DISCUSSION THREAD

**Discussion by Cldndan** [wixtoolset/issues#7497] [Questions]
I want to distribute a total of three files for example.
(Three files called xxx.exe, yyy.dll, and zzz.dll.)

And I'll let you know what I know to make sure I'm right or wrong.
We know that the above three files are compressed and mounted in xxx.msi for example.

The requirements are:

1. I want to deploy the file under C:\xxx of the target system.
2. Accordingly, it must be able to be changed through the input UI.
3. The executable file, xxx.exe, should be deployed to C:\xxx, the path specified by 1 and 2.
4. The following requirements are additionally required for the two dll files, yyy.dll and zzz.dll.
- It is necessary to review whether the dll files loaded in the msi are in the system folder of the target system. This is a follow-up requirement.
1) If the dll files loaded with msi are newer versions than those of the target system or older versions, they must be distributed to C:\xxx.
-This is to minimize the impact on other programs running in the target system.
2) If the dll files loaded with the msi have the same version as the target system, do not distribute them.
- Don't waste resources on storage space on the target system!
3) If the dll files loaded with msi are not in the target system, try to distribute the dll files to the system folder.

5. The installer reviews the deployed state and has the following additional requirements.
1) If it is not distributed, a new distribution button is required.
2) If it is already deployed, the overwrite button is required.
- Must meet the requirements of 1 to 4.
- A delete button is required.
- This is the same as the normal installer.

I tried to write Wix XML, but only XML that is unconditionally distributed to c:\xxx is implemented.

**Comment by BMurri** [comment]
Some things to consider WRT the two dll files:
Over the past couple of decades, Microsoft has been locking down the system folder in increasingly aggressive ways. Additionally, today many anti-virus solutions will interfere with programs using newly added DLLs that weren't placed there by the operating system's own updaters (of which Windows Installer is not on the list).

If you do succeed in deploying one or both of those files to the system folder, there's no way to guarantee preventing the operating system from later updating those files to other versions. If your xxx.exe really requires exact versions of those two dlls, the only correct way to deploy is to put them in the same directory as the executable.

TL/DR: don't change the system folder. Ever.

**Comment by Cldndan** [comment]
yes i know what you mean.

First of all, the dll file I suggested was used as an example, so I used a pseudonym instead of the file name.

In other words, the dlls I want to distribute are dll files officially distributed by Windows or Microsoft.

However, due to the configuration of the target system, I cannot guarantee that the dll required for the executable file I developed exists in the target system.

Also, the same dll is distributed to prevent waste of storage space and waste of registry.

We know your concerns.
In other words, it is judged that it is okay to distribute to the system folder through automatic upgrade of Windows.
The reason is that part of the requirements I requested, the requirement to deploy the dll file to the target system, is to deploy it to the system folder only when the dll file does not exist.
In addition, it is judged that there is no concern due to the following reasons.

1. After upgrading, if a problem occurs in the program I made, redistribution will solve the problem.
 If all my requirements are met, the dll loaded in the msi will be the latest or old version, so it will be distributed in the same path as the executable file. In other words, after the problem occurs, the problem will be eliminated by redistribution, that is, by post-processing, so I would like you to write what I requested.

Even if you look at the manual, it was difficult to find the example or keyword you want to refer to, and if you do not know the majority of tag names, it is difficult to search, so we politely request.

**Reply by BMurri** [reply]
If the DLLs are officially distributed by their owners, then if you want them written to the system folder, you should always use the owners' installers. That's one of the big reasons WiX also provides "Burn", the bootstrapper/chainer.

**Reply by BMurri** [reply]
The other big reason is because Windows Installer UI is **really bad** and  most of us are able to make much better UIs that work in more places.

**Reply by BMurri** [reply]
If you can legally redistribute the DLLs to your application's installation directory, feel free to always do that and don't worry about the versions in the system folder.
